src/WeakValueDict.jl

```diff
@@ -145,26 +511,34 @@ function Base.get(wkh::WeakValueDict{K, V}, key, default) where {K, V}
         if x === default
             return x
         else
-            return x.value::V
+            y = x.value
+            if y === nothing
+                return default
+            else
+               return y::V
+            end
         end
     end
 end
+
+#### the rest of these look dodgy
+
 function Base.get(default::Base.Callable, wkh::WeakValueDict{K}, key) where {K}
     key === nothing && throw(KeyError(nothing))
     lock(wkh) do
-        return get(default, wkh.ht, key)
+        return get(default, wkh.ht, key)  # ???
     end
 end
 function Base.pop!(wkh::WeakValueDict{K}, key) where {K}
     key === nothing && throw(KeyError(nothing))
     lock(wkh) do
-        return pop!(wkh.ht, key).value
+        return pop!(wkh.ht, key).value  # ???
     end
 end
 function Base.pop!(wkh::WeakValueDict{K}, key, default) where {K}
     key === nothing && return default
     lock(wkh) do
-        return pop!(wkh.ht, key, default)
+        return pop!(wkh.ht, key, default)  # ect.
```

Comment:
Do you mean "etc." ? But what is this comment supposed to tell me?

Reply:
So how about
```suggestion
        return pop!(wkh.ht, key, default)  # FIXME: this returns a WeakRef, but shouldn't
```
or
```suggestion
        return pop!(wkh.ht, key, default).value
```
And if `nothing` is still possible at this stage, then of course that needs to be checked for and `default` returned.